streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java

```diff
@@ -67,6 +68,58 @@
     }
 
     @Override
+    public KeyValueIterator<Bytes, byte[]> all() {
+        final List<Segment> searchSpace = keySchema.allSegments(segments);
+        
+        final Segment minSegment = keySchema.minSegment(segments);
```

Comment:
can we just use `segments.minSegment()` here? and `segments.maxSegment()` below?